xbmc/games/controllers/input/PortInput.cpp

```diff
@@ -40,7 +40,7 @@ void CPortInput::RegisterInput(JOYSTICK::IInputProvider* provider)
 void CPortInput::UnregisterInput(JOYSTICK::IInputProvider* provider)
 {
   // Unregister in reverse order
-  if (provider == nullptr)
+  if (provider == nullptr && m_appInput)
```

Comment:
`m_appInput != nullptr`?

Reply:
`m_appInput` is a shared pointer and I'd rather the pointer unwrapping take place in `std::unique_ptr::operator bool()`